scripts/compile/common.sh

```diff
@@ -133,6 +136,7 @@ $CFLAGS_BUILD_TYPE \
 "
 
 COMMON_CXXFLAGS=" \
+-std=c++2b \
```

Comment:
```suggestion
-std=c++20 \
```

could we align the c++ version with react-native?

Reply:
tthe `2b` was a mistake, I meant c++23. this version of WebKit/JSC doesn't demand c++23, so c++20 should be fine.

Reply:
Adding in the c++20 flags explicitly, I can see that WebKit itself is still sprinkling in  `-std=c++2b` in several places. I'll try and patch that out, and hope that it doesn't cause a cascade of new issues. If it does, then we'll need c++23 and NDK 28 to move forward with the Bun alignment.

Reply:
> If it does, then we'll need c++23 and NDK 28 to move forward with the Bun alignment.

Jumping here to say that RN is not on NDK 28/C++23 yet. Prebuilding JSC with a version ahead of RN could lead to runtime problems

Reply:
> > If it does, then we'll need c++23 and NDK 28 to move forward with the Bun alignment.
> 
> Jumping here to say that RN is not on NDK 28/C++23 yet. Prebuilding JSC with a version ahead of RN could lead to runtime problems

this is good to know. in the worst case, I'll keep the repo forked since my deployment into BabylonNative isn't held back by the constraints of React Native. when React Native chooses to catch up, the PR will be ready. since most of the React Native community was funneled onto Hermes, maybe it isn't a big deal? :)

Reply:
> since most of the React Native community was funneled onto Hermes, maybe it isn't a big deal?

Yeah exactly. I believe we assume less than 1% of the users are on JSC at this stage